nav2_docking/opennav_docking/src/simple_charging_dock.cpp

```diff
@@ -105,6 +116,26 @@ void SimpleChargingDock::configure(
   node_->get_parameter(name + ".staging_x_offset", staging_x_offset_);
   node_->get_parameter(name + ".staging_yaw_offset", staging_yaw_offset_);
 
+  node_->get_parameter(name + ".detector_service_name", detector_service_name_);
+  node_->get_parameter(name + ".detector_service_timeout", detector_service_timeout_);
+  node_->get_parameter(name + ".subscribe_toggle", subscribe_toggle_);
+
+  // Initialize detection state
+  detection_started_ = false;
```

Comment:
Rename here and in the noncharging plugin to be `detection_active_` to be more clear